### PR TITLE
fix: rich select issues

### PR DESCRIPTION
**Pull request by DorianMaliszewski**
## Type

- Bug

#### What is expected?

- When in multi mode, select doesn't display the label
- When isSearchable, select display an input which is not vertical aligned
- When selecting an option, select doesn't keep the scroll (documented)
- When disabled, select doesn't display the label

#### The following changes where made:

1. Lot of changes in RichSelect styles and props

2. UnitInput test wasn't good so fix it and comment

3. Add isSearchable story

### CODE REVIEW

File: src/components/UnitInput/__tests__/index.js [hunk 1]
```diff
@@ -80,13 +80,21 @@ describe('UnitInput', () => {
     )
   })
 
-  test(`renders with RichSelect update`, async () => {
+  test.only(`renders with RichSelect update`, async () => {
```
Comment (chambo-e) [inline src/components/UnitInput/__tests__/index.js:83]: 💩

**Review by matthprost — CHANGES_REQUESTED** [review]
Searcheable test is hovering label a bit:

https://user-images.githubusercontent.com/15812968/119318614-d2e1a500-bc79-11eb-9585-2b0af1485535.mov

File: src/components/UnitInput/__tests__/index.js (hunk 1 above)
Comment (DorianMaliszewski) [reply src/components/UnitInput/__tests__/index.js:83]: Oups

File: package.json [hunk 2]
```diff
@@ -66,7 +66,8 @@
     ],
     "coverageReporters": [
       "text",
-      "cobertura"
+      "cobertura",
+      "html"
```
Comment (chambo-e) [inline package.json:70]: Useless noise no ? We can add the reporter whenever we want with the cli

File: src/components/RichSelect/__tests__/index.js [hunk 3]
```diff
@@ -20,12 +20,18 @@ const customStyles = {
...
         <RichSelect.Option value="b">Option B</RichSelect.Option>
       </RichSelect>,
+      {
+        transform: async node => {
+          const input = node.getByRole('textbox')
+          userEvent.click(input)
+        },
+      },
     )
   })
```
Comment (chambo-e) [inline src/components/RichSelect/__tests__/index.js:36]:
```suggestion
  test('renders correctly uncontrolled', () => (
    shouldMatchEmotionSnapshot(
      <RichSelect inputId="test" labelId="test-label" name="uncontrolled">
        <RichSelect.Option value="a">Option A</RichSelect.Option>
        <RichSelect.Option value="b">Option B</RichSelect.Option>
      </RichSelect>,
      {
        transform: async node => {
          const input = node.getByRole('textbox')
          userEvent.click(input)
        },
      },
    )
  ))
```

**Review by chambo-e — CHANGES_REQUESTED** [review]

File: src/components/Tags/index.js [hunk 4]
```diff
@@ -194,7 +194,7 @@ const Tags = ({
                 : 'base'
             }
             disabled={disabled}
-            key={`tag-${tag.index}`}
+            key={`${tag.index}`}
```
Comment (matthprost) [inline src/components/Tags/index.js:197]:
```suggestion
            key={tag.index}
```

**Review by matthprost — CHANGES_REQUESTED** [review]

**Review by matthprost — APPROVED** [review]

File: src/components/Tags/index.js (hunk 4 above)
Comment (vincentaudebert) [reply src/components/Tags/index.js]: FYI, j'ai rajoute le label dans ma MR car le UUID est pas unique pour tous les tags

**Comment by vincentaudebert** [comment]
Screenshot 2021-05-27 at 11 35 15


Design suggestion but should we align center when searching?

File: src/components/Tags/index.js (hunk 4 above)
Comment (DorianMaliszewski) [reply src/components/Tags/index.js]: Je peux quand même laisser où ca te derange ?

Comment (vincentaudebert) [reply src/components/Tags/index.js]: nah all good cetait juste pour info :)

File: src/components/RichSelect/__tests__/index.js (hunk 3 above)
Comment (chambo-e) [reply src/components/RichSelect/__tests__/index.js]: You need to return the shouldMatchEmotionSnapshot for the await to be triggered

File: src/components/Tags/index.js [hunk 5]
```diff
@@ -194,7 +194,7 @@ const Tags = ({
                 : 'base'
             }
             disabled={disabled}
-            key={`tag-${tag.index}`}
+            key={tag.index}
```
Comment (chambo-e) [inline src/components/Tags/index.js:197]: This is too much conflict prone no ?

Comment (DorianMaliszewski) [reply src/components/Tags/index.js:197]: This key is randomly generated

**Review by chambo-e — APPROVED** [review]

**Pull request merged**